src/babel/index.ts

```diff
@@ -0,0 +1,177 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import { NodePath, PluginObj } from '@babel/core';
+import * as BabelTypes from '@babel/types';
+const PACKAGE_NAME = 'react-loosely-lazy';
+const BUNDLER_CACHE_ID_KEY = 'getCacheId';
+const MODULE_ID_KEY = 'moduleId';
+const LAZY_METHODS = ['lazyForPaint', 'lazyAfterPaint', 'lazy'];
+
+const DEFAULT_OPTIONS: {
+  [key: string]: { ssr: boolean; defer: number };
+} = {
+  lazyForPaint: { ssr: true, defer: 0 },
+  lazyAfterPaint: { ssr: true, defer: 1 },
+  lazy: { ssr: false, defer: 2 },
+};
+
+export default function({
+  types: t,
+  template,
+}: {
+  types: typeof BabelTypes;
+  template: any;
+}): PluginObj {
+  return {
+    visitor: {
+      ImportDeclaration(
+        path: NodePath<BabelTypes.ImportDeclaration>,
+        state: { opts?: { client?: boolean } }
+      ) {
+        const { client } = state.opts || {};
+        const source = path.node.source.value;
+
+        if (source !== PACKAGE_NAME) {
+          return;
+        }
+
+        const bindingNames = LAZY_METHODS;
+        const bindings = bindingNames
+          .map(name => path.scope.getBinding(name))
+          .filter(binding => binding);
+
+        bindings.forEach((binding: any) => {
+          const lazyMethodName = binding.identifier.name;
+
+          binding.referencePaths.forEach((refPath: any) => {
+            let callExpression = refPath.parentPath;
+
+            if (
+              callExpression.isMemberExpression() &&
+              callExpression.node.computed === false &&
+              callExpression.get('property').isIdentifier({ name: 'Map' })
+            ) {
+              callExpression = callExpression.parentPath;
+            }
+
+            if (!callExpression.isCallExpression()) {
+              return;
+            }
+
+            const args = callExpression.get('arguments');
+
+            if (!args.length) {
+              throw callExpression.error;
+            }
+
+            const lazyImport = args[0];
+            let lazyOptions = args[1];
+
+            // ensures that options exist even if not passed explicitly
+            if (!lazyOptions || !lazyOptions.isObjectExpression()) {
+              callExpression.node.arguments.push(t.objectExpression([]));
+              lazyOptions = callExpression.get('arguments')[1];
+            }
+
+            if (!lazyImport.isFunction()) {
+              return;
+            }
+
+            let lazyImportPath = null;
+
+            lazyImport.traverse({
+              Import(importPath: NodePath<BabelTypes.ImportDeclaration>) {
+                lazyImportPath = importPath.parentPath;
+              },
+            });
+
+            if (!lazyImportPath) {
```

Comment:
When will this happen? Should we throw an error instead if something is not transformed or configured as expected?

Reply:
I don't know if this plugin should be responsible for throwing errors in response to your app code. You may have called the lazy function incorrectly... but is it this plugins responsibility to explode if so?